packages/server/src/utilities/workerRequests.ts

```diff
@@ -10,29 +16,61 @@ import {
 } from "@budibase/backend-core"
 import { Ctx, User, EmailInvite } from "@budibase/types"
 
-export function request(ctx?: Ctx, request?: any) {
-  if (!request.headers) {
-    request.headers = {}
+function ensureHeadersIsObject(headers: HeadersInit | undefined): Headers {
+  if (headers instanceof Headers) {
+    return headers
+  }
+
+  const headersObj = new Headers()
+  if (headers === undefined) {
+    return headersObj
   }
 
+  if (Array.isArray(headers)) {
+    for (const [key, value] of headers) {
+      headersObj.append(key, value)
+    }
+  } else {
+    for (const key in headers) {
+      headersObj.append(key, headers[key])
+    }
+  }
+  return headersObj
+}
+
+export function request(request: RequestInit & { ctx?: Ctx }): RequestInit {
+  const ctx = request.ctx
+  request.headers = ensureHeadersIsObject(request.headers)
+
   if (!ctx) {
-    request.headers[constants.Header.API_KEY] = coreEnv.INTERNAL_API_KEY
+    if (coreEnv.INTERNAL_API_KEY) {
+      request.headers.set(constants.Header.API_KEY, coreEnv.INTERNAL_API_KEY)
```

Comment:
I'm not sure this is strictly correct for the type `request` actually is - its a request to node fetch in which the headers are simply an object.

Reply:
The type for `request.headers` is `HeadersInit` which is defined here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b2c721481ee38f65eb6630e958cc0c3383c81831/types/node-fetch/index.d.ts#L194. It's a bit wider than just an object, so I created the `ensureHeadersObject` function to make sure it's always a `Headers` object, and used that throughout. Wasn't sure about this as I was doing it but it gets the type checking the pass, and handles the wider definition of what headers can be.

Reply:
`node-fetch` has a very weird Typescript definition here, so what is happening:
1. When you are building a request to be made by node-fetch, generally you need to just use a map, as your input is simply JSON e.g.
```fetch({
  headers: {...}
})```
This is what we are building towards, so I'm not sure the `.set` functions will be available.
2. When a node-fetch response is received it will come back in the format described by `Headers` - its inconsistent and very frustrating, but it does happen that way.